app/controlplane/internal/service/workflowcontract.go

```diff
@@ -91,11 +94,17 @@ func (s *WorkflowContractService) Create(ctx context.Context, req *pb.WorkflowCo
 		return nil, err
 	}
 
+	token, err := usercontext.GetRawToken(ctx)
+	if err != nil {
+		return nil, status.Error(codes.Unauthenticated, "invalid token")
+	}
+
 	// Currently supporting only v1 version
 	schema, err := s.contractUseCase.Create(ctx, &biz.WorkflowContractCreateOpts{
 		OrgID: currentOrg.ID,
 		Name:  req.Name, Description: req.Description,
-		RawSchema: req.RawContract})
+		RawSchema: req.RawContract,
+		Token:     token})
```

Comment:
I find strange passing the token here, it looks like it's going to get stored.

What about creating another method in the use-case called `ValidateContractPolicies` and call it maybe at the service level? I am not 100 sure about it either but at least this "optional" feature is not part of the core part of creating/updating a contract. WDYT?

Reply:
I know what you mean. I'll try extracting the logic to its own use case method. It might be cleaner and easier to understand.